### PR TITLE
[Bug](sort)fix merge sorter might div zero when block bytes less than block rows

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #xxx
If block bytes are bigger than the corresponding block's rows, then the avg_size_per_row would be zero. Which would end up diving zero in the following logic.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/suites/demo_p0/merge_sort.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    qt_sql """
+        SELECT subq_0.`c1` AS c1
+        FROM
+        (SELECT version() AS c0,
+                  ref_0.`id` AS c1
+        FROM test_streamload_action1 AS ref_0
+        LEFT JOIN B AS ref_9 ON (ref_0.`id` = ref_9.`b_id`)
+        WHERE ref_9.`b_id` IS NULL) AS subq_0
+        WHERE subq_0.`c0` IS NOT NULL
+        ORDER BY subq_0.`c0` DESC
```
Comment (jacktengg) [inline regression-test/suites/demo_p0/merge_sort.groovy:42]: need add another other by to ensure test result stability.

Comment (ByteYue) [reply regression-test/suites/demo_p0/merge_sort.groovy:42]: > need add another order by to ensure test result stability.

OK, I'll refresh the regression case after I've rebuilt the be binary and run the case.

**Comment by jacktengg** [comment]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.29 seconds
 load time: 560 seconds
 storage size: 17123097128 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230112095621_clickbench_pr_78803.html

**Comment by jacktengg** [comment]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**